satellite/gracefulexit/endpoint.go

```diff
@@ -204,6 +204,15 @@ func (endpoint *Endpoint) doProcess(stream processStream) (err error) {
 		endpoint.connections.delete(nodeID)
 	}()
 
+	// check if node is disqualified
+	nodeInfo, err := endpoint.overlay.Get(ctx, nodeID)
+	if err != nil {
+		return rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
+	}
+	if nodeInfo.Disqualified != nil {
+		return rpcstatus.Error(rpcstatus.PermissionDenied, "Only undisqualified node allowed for graceful exit")
```

Comment:
should we go ahead and update these nodes to have completed + failed graceful exit? Otherwise couldn't they remain as "gracefully exiting" and never finish because of this error?